code/game/objects/structures/aliens.dm

```diff
@@ -82,23 +96,150 @@
 /obj/structure/alien/resin/wall/BlockSuperconductivity()
 	return TRUE
 
-/obj/structure/alien/resin/membrane
-	name = "resin membrane"
-	desc = "Resin just thin enough to let light pass through."
-	icon = 'icons/obj/smooth_structures/alien/resin_membrane.dmi'
-	icon_state = "resin_membrane-0"
-	base_icon_state = "resin_membrane"
-	opacity = FALSE
-	max_integrity = 160
-	resintype = "membrane"
+/*
+ *Resin-Door - Borrows its code from Mineral-Door, not a subtype due to needing many overrides if so
+*/
+/obj/structure/alien/resin/door
```

Comment:
this is pretty gross, is there not a great way to just use subtypes here?

Reply:
I've asked a few people about this, because this first /was/ a subtype, but the issue with this is;
If I make it a subtype of mineral doors, I need to first override most of the functionalities of the above (Borgs can open them, they need to close automatically, their sounds are wrong, half their variables are simply unneded for xeno resin door) alongside adding extra overrides because, as they are not /resin/ but /mineral_door/ they don't inherit the /resin/ stuff, so I'd have to re-do most things, like attack modifier of brute/burn, sounds, behavior of can_open, etc. Specially since the door itself also needs a mobless try_to_operate for its callbacks to close automatically in the way the doors are to function. 

It /is/ very bleh but Mineral_doors don't have a subtype above them of simple doors or such, and the old code is not really good for it. If you'd like me to detail further I can hit you up in DM's about it?